src/core/multifeaturelistmodelbase.cpp

```diff
@@ -543,3 +640,32 @@ void MultiFeatureListModelBase::attributeValueChanged( QgsFeatureId fid, int idx
     }
   }
 }
+
+void MultiFeatureListModelBase::geometryChanged( QgsFeatureId fid, const QgsGeometry &geometry )
+{
+  QgsVectorLayer *l = qobject_cast<QgsVectorLayer *>( sender() );
+  Q_ASSERT( l );
+
+  int i = 0;
+  for ( auto &pair : mFeatures )
```

Comment:
```suggestion
  for ( auto &pair : qgis::as_const( mFeatures ) )
```

or can the method be made const?

Reply:
Can it be const if we are modifying list items?

Reply:
Can this (or the method) be const if we're modifying lists elements?